api_tests/preprints/views/test_preprint_detail.py

```diff
@@ -739,68 +716,6 @@ def test_cannot_set_primary_file(self, app, user, preprint, url):
             expect_errors=True)
         assert res.status_code == 403
 
-    def test_write_contribs_can_set_subjects(
-            self, app, user, preprint, subject, url):
-
-        # def test_write_contrib_can_set_subjects(self, app, user, preprint,
-        # subject, url):
-        write_contrib = AuthUserFactory()
-        preprint.add_contributor(
-            write_contrib,
-            permissions='write',
-            auth=Auth(user), save=True)
-
-        assert not preprint.subjects.filter(_id=subject._id).exists()
-        update_subjects_payload = build_preprint_update_payload(
-            preprint._id, attributes={'subjects': [[subject._id]]})
-
-        res = app.patch_json_api(
-            url, update_subjects_payload,
-            auth=write_contrib.auth,
-            expect_errors=True)
-        assert res.status_code == 200
-
-        assert preprint.subjects.filter(_id=subject._id).exists()
-
-    def test_contribs_cannot_set_subjects(
-            self, app, user, preprint, subject, url):
-        # def test_read_contrib_can_set_subjects(self, app, user, preprint,
-        # subject, url):
-        read_contrib = AuthUserFactory()
-        preprint.add_contributor(
-            read_contrib,
-            permissions='read',
-            auth=Auth(user), save=True)
-
-        assert not preprint.subjects.filter(_id=subject._id).exists()
-        update_subjects_payload = build_preprint_update_payload(
-            preprint._id, attributes={'subjects': [[subject._id]]})
-
-        res = app.patch_json_api(
-            url, update_subjects_payload,
-            auth=read_contrib.auth,
-            expect_errors=True)
-        assert res.status_code == 403
-
-        assert not preprint.subjects.filter(_id=subject._id).exists()
-
-    # def test_non_contrib_cannot_set_subjects(self, app, user, preprint,
-    # subject, url):
-        non_contrib = AuthUserFactory()
-
-        assert not preprint.subjects.filter(_id=subject._id).exists()
-
-        update_subjects_payload = build_preprint_update_payload(
-            preprint._id, attributes={'subjects': [[subject._id]]})
-
-        res = app.patch_json_api(
-            url, update_subjects_payload,
-            auth=non_contrib.auth,
-            expect_errors=True)
-        assert res.status_code == 403
-
-        assert not preprint.subjects.filter(_id=subject._id).exists()
```

Comment:
testing updating subjects on the resource directly has been consolidated on the UpdateSubjectsMixin